MAUI/DataGrid/filtering.md

```diff
@@ -0,0 +1,184 @@
+                return false;
+            }
+            else
+            {
+                if (!CheckEquals(orderInfo.OrderID.ToString()) ||
+                   !CheckEquals(orderInfo.CustomerID) ||
+                   !CheckEquals(orderInfo.Customer) ||
+                   !CheckEquals(orderInfo.ShipCountry) ||
+                   !CheckEquals(orderInfo.ShipCity))
+                    return true;
+                return false;
+            }
+        }
+        else
+        {
+            var value = record.GetType().GetProperty(columns.SelectedItem.ToString().Replace(" ",""));
+            var exactValue = value.GetValue(record, null);
+            if (conditions.SelectedItem.ToString() == "Contains")
+            {
+                return FilterText.ToLower().Contains(exactValue.ToString().ToLower());
+            }
+            else if (conditions.SelectedItem.ToString() == "Equals")
+            {
+                return CheckEquals(exactValue.ToString());
+            }
+            else
+            {
+                return !CheckEquals(exactValue.ToString());
+            }
+        }
+    }
+    return false;
+}
+
+public bool CheckEquals(string value)
+{
+    return FilterText.Equals(value);
+}
+
+private void Button_Clicked(object sender, EventArgs e)
+{
+	this.sfDataGrid.View.Filter = FilterRecords;
+	this.sfDataGrid.View.RefreshFilter();
+}
+{% endhighlight %}
+{% endtabs %}
+
+The following code example illustrates how to create a [Picker](https://docs.microsoft.com/en-us/dotnet/maui/user-interface/controls/picker) for conditions and add appropriate strings to that Picker and how the records will be filtered based on selected conditions,
+
+{% tabs %}
+{% highlight xaml %}
+<Grid Grid.Row="1" Grid.Column="0">
+    <Grid.ColumnDefinitions>
+        <ColumnDefinition Width="150"/>
+        <ColumnDefinition Width="150"/>
+        <ColumnDefinition Width="150"/>
+    </Grid.ColumnDefinitions>
+
+    <Picker
+        x:Name="columns"
+        Grid.Row="0" Grid.Column="0">
+        <Picker.Items>
+            <x:String>All Columns</x:String>
+            <x:String>Order ID</x:String>
+            <x:String>Customer ID</x:String>
+            <x:String>Customer</x:String>
+            <x:String>Ship City</x:String>
+            <x:String>Ship Country</x:String>
+        </Picker.Items>
+        <Picker.SelectedItem>
+            <x:String>All Columns</x:String>
+        </Picker.SelectedItem>
+    </Picker>
+    <Picker 
+        x:Name="conditions"
+        Grid.Row="0" Grid.Column="1">
+        <Picker.Items>
+            <x:String>Equals</x:String>
+            <x:String>Does Not Equal</x:String>
+            <x:String>Contains</x:String>
+        </Picker.Items>
+    </Picker>
+
+    <Button Grid.Row="0" Grid.Column="3" Text="Filter" Clicked="Button_Clicked"/>
+</Grid>
+{% endhighlight %}
+{% endtabs %}
+
+## Clear filtering
+
+You can clear the applied filtering by setting the `SfDataGrid.View.Filter` property to `null`.

```

Comment:
Clear the applied filtering by setting the `SfDataGrid.View.Filter` property to `null.`

Reply:
Modified